CodeEdit/Features/QuickOpen/ViewModels/FuzzySearch.swift

```diff
@@ -0,0 +1,203 @@
+//
+//  FuzzySearch.swift
+//  CodeEdit
+//
+//  Created by Tommy Ludwig on 04.05.23.
+//
+
+import Foundation
+
+enum FuzzySearch {
+    /// Searches an array of view models for occurrences of a fuzzy search query.
+    ///
+    /// This function takes a fuzzy search `query` and an array of `URL`s, and returns a new array that contains only
+    /// those url's that match the query.
+    /// The function uses the `score` function to calculate a score for each url and
+    /// includes only those url's whose scores are greater than 0.0.
+    /// The resulting array is then sorted by a score, in descending order.
+    ///
+    /// - Parameters:
+    ///   - query: A `String` value representing the fuzzy search query.
+    ///   - urls: An array of `URL`s, each representing a file, to search within.
+    /// - Returns: An array of `URL`s that match the fuzzy search query, sorted by score.
+    static func search(query: String, in urls: [URL]) -> [URL] {
+        let filteredResult = urls.filter { url -> Bool in
+            let nameScore = score(query: query, url: url)
+            return nameScore > 0.0
+        }
+
+        let sortedResult = filteredResult.sorted { url1, url2 -> Bool in
+            let nameScore1 = score(query: query, url: url1)
+            let nameScore2 = score(query: query, url: url2)
+            if nameScore1 > nameScore2 {
+                return true
+            } else {
+                return false
+            }
```

Comment:
This can be shorter written as `return nameScore1 > nameScore2`

Reply:
Thanks for the tip 🙏